src/server/dfly_main.cc

```diff
@@ -494,17 +603,30 @@ Usage: dragonfly [FLAGS]
     }
   }
 
+  auto memory = ReadMemInfo().value();
```

Comment:
1. If ReadMemInfo() returns an error - you will get an exception. We do not rely on exceptions. Please at least check-fail before accessing `.value()`. 
2. I would combine `InsideContainer and ReadContainerMemoryLimits` functions together into `MergeContainerLimitsIfNeeded`  (shorter name is fine too 😄 ).

Reply:
About 1 - before adding this change we still did not check (see: https://github.com/dragonflydb/dragonfly/blob/246f6093db89f1b14e63c20b1ce6cf33e7f2accd/src/server/dfly_main.cc#L500-L501). I did find this odd, but assumed it makes sense as theoretically this should never fail.

About 2 - sounds great, would also avoid global `cgroup` variable.